src/main/java/uk/gov/hmcts/cmc/claimstore/services/ccd/callbacks/caseworker/ChangeContactDetailsNotificationService.java

```diff
@@ -0,0 +1,110 @@
+import uk.gov.hmcts.reform.ccd.client.model.AboutToStartOrSubmitCallbackResponse;
+import uk.gov.hmcts.reform.ccd.client.model.CallbackResponse;
+
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Map;
+
+import static uk.gov.hmcts.cmc.claimstore.services.notifications.content.NotificationTemplateParameters.CLAIMANT_NAME;
+import static uk.gov.hmcts.cmc.claimstore.services.notifications.content.NotificationTemplateParameters.CLAIM_REFERENCE_NUMBER;
+import static uk.gov.hmcts.cmc.claimstore.services.notifications.content.NotificationTemplateParameters.DEFENDANT_NAME;
+import static uk.gov.hmcts.cmc.claimstore.services.notifications.content.NotificationTemplateParameters.EXTERNAL_ID;
+import static uk.gov.hmcts.cmc.claimstore.services.notifications.content.NotificationTemplateParameters.FRONTEND_BASE_URL;
+
+@Service
+public class ChangeContactDetailsNotificationService {
+
+    private static final String ERROR_MESSAGE =
+        "There was a technical problem. Nothing has been sent. You need to try again.";
+    private final NotificationService notificationService;
+    private final NotificationsProperties notificationsProperties;
+    private final CaseDetailsConverter caseDetailsConverter;
+    private static final Logger logger = LoggerFactory.getLogger(LoggerHandler.class);
+
+    @Autowired
+    public ChangeContactDetailsNotificationService(
+        CaseDetailsConverter caseDetailsConverter,
+        NotificationService notificationService,
+        NotificationsProperties notificationsProperties
+    ) {
+        this.notificationService = notificationService;
+        this.notificationsProperties = notificationsProperties;
+        this.caseDetailsConverter = caseDetailsConverter;
+    }
+
+    public CallbackResponse sendEmailToRightRecipient(CCDCase ccdCase, Claim claim) {
+        boolean errors = false;
+        try {
+
+            if (ccdCase.getContactChangeParty() == CCDContactPartyType.CLAIMANT) {
+                notifyDefendant(claim);
+            } else {
+                notifyClaimant(claim);
+            }
+        } catch (Exception e) {
+            logger.info("Sending email to party failed", e);
+            errors = true;
+        }
+        if (!errors) {
+            logger.info("Change Contact Details: Email was sent, case is being updated");
+            CCDCase updatedCase = ccdCase.toBuilder()
+                .contactChangeParty(null)
+                .contactChangeContent(null)
+                .generalLetterContent(null)
+                .build();
+            return AboutToStartOrSubmitCallbackResponse
+                .builder()
+                .data(caseDetailsConverter.convertToMap(updatedCase))
+                .build();
+        } else {
+            return AboutToStartOrSubmitCallbackResponse
+                .builder()
+                .errors(Collections.singletonList(ERROR_MESSAGE))
+                .build();
+        }
+    }
```

Comment:
When returning an exception this is a better pattern to follow as it avoids nested ifs in the future and puts the happy path return object at the end of the method

```suggestion
        if (errors) {
              return AboutToStartOrSubmitCallbackResponse
                .builder()
                .errors(Collections.singletonList(ERROR_MESSAGE))
                .build();
        }
        
        logger.info("Change Contact Details: Email was sent, case is being updated");
        CCDCase updatedCase = ccdCase.toBuilder()
              .contactChangeParty(null)
              .contactChangeContent(null)
               generalLetterContent(null)
               build();
          return AboutToStartOrSubmitCallbackResponse
              .builder()
              .data(caseDetailsConverter.convertToMap(updatedCase))
              .build();
    }
```

Reply:
I have removed the usage off errors flag